lib/src/link/auth/link_auth.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:async';
+
+import 'package:graphql_flutter/src/link/link.dart';
+import 'package:graphql_flutter/src/link/operation.dart';
+import 'package:graphql_flutter/src/link/fetch_result.dart';
+
+typedef GetToken = String Function();
+
+class AuthLink extends Link {
+  AuthLink({
+    this.getToken,
+  }) : super(
+          request: (Operation operation, [NextLink forward]) {
+            StreamController<FetchResult> controller;
+
+            Future<void> onListen() async {
+              try {
+                final String token = getToken();
```

Comment:
Just checking in - is there a way here to also support async token calls?

Particularly in my use case (maybe others as well), I need to support cases where the token gets expired and needs to be refreshed and that would require doing the token call asynchronously.

Reply:
I think #178 should solve this, if not go ahead and open a new issue